demisto_sdk/commands/test_content/test_use_case/test_use_case.py

```diff
@@ -0,0 +1,661 @@
+        if self.client:
+            self.client.api_client.pool.close()
+            self.client.api_client.pool.terminate()
+            del self.client
+        self.client = demisto_client.configure(
+            base_url=self.base_url,
+            api_key=self.api_key,
+            auth_id=self.auth_id,
+            verify_ssl=False,
+        )
+
+    def execute_tests(self):
+        try:
+            self.build_context.logging_module.info(
+                f"Starts tests with server url - {get_ui_url(self.ui_url)}",
+                real_time=True,
+            )
+            start_time = get_utc_now()
+            self.build_context.logging_module.info(
+                f"Running the following tests: {self.tests}",
+                real_time=True,
+            )
+
+            cloud_client = get_client_from_server_type(
+                base_url=self.base_url, api_key=self.api_key, auth_id=self.auth_id
+            )
+
+            for i, test_use_case_directory in enumerate(self.tests, start=1):
+                logger.info(
+                    f"<cyan>[{i}/{len(self.tests)}] test use cases: {get_relative_path_to_content(test_use_case_directory)}</cyan>",
+                )
+
+                success, test_use_case_test_suite = run_test_use_case_pytest(
+                    test_use_case_directory,
+                    cloud_client=cloud_client,
+                    lcas_id=self.build_context.lcas_id,
+                )
+
+                if success:
+                    logger.info(
+                        f"<green>Test use case {get_relative_path_to_content(test_use_case_directory)} passed</green>",
+                    )
+                else:
+                    self.build_context.tests_data_keeper.errors = True
+                    logger.error(
+                        f"<red>Test use case {get_relative_path_to_content(test_use_case_directory)} failed</red>",
+                    )
+                if test_use_case_test_suite:
+                    test_use_case_test_suite.add_property(
+                        "start_time",
+                        start_time,  # type:ignore[arg-type]
+                    )
+                    self.build_context.tests_data_keeper.test_results_xml_file.add_testsuite(
+                        test_use_case_test_suite
+                    )
+
+                    self.build_context.logging_module.info(
+                        f"Finished tests with server url - " f"{self.ui_url}",
+                        real_time=True,
+                    )
+            duration = duration_since_start_time(start_time)
+            self.build_context.logging_module.info(
+                f"Finished tests with server url - {self.ui_url}, Took: {duration} seconds",
+                real_time=True,
+            )
+        except Exception:
+            self.build_context.logging_module.exception("~~ Thread failed ~~")
+            self.build_context.tests_data_keeper.errors = True
```

Comment:
how does it behave in case of exception?
1. will the rest of the threads continue?
2. will we see the error message?

Reply:
yes, each thread should run its tests, the rest of the threads should keep running.
Also logs per thread are printed (with thread number)